proposals/OpaqueReflect.md

```diff
@@ -0,0 +1,132 @@
+# Specification of opaque reflection
+
+## Goal
+
+This PR addresses #1892. The goal is to permit the reflection of functions which contain symbols that are not already in the logic, through the automatic lifting of the symbols which are not in the logic, using measures.
+
+## Syntax and use
+
+### First, real-world example
+
+```Haskell
+{-@ LIQUID "--reflection"      @-}
+
+module OpaqueRefl04 where
+
+{-@ reflect keepEvens @-}
+keepEvens :: [Int] -> [Int]
+keepEvens = filter even
+```
+
+It will introduce measures for filter and even, such that the result would be equivalent to
+
+```Haskell
+{-@ LIQUID "--reflection"      @-}
+
+module OpaqueRefl06 where
+
+{-@ measure GHC.List.filter :: (a -> Bool) -> [a] -> [a] @-}
+{-@ assume GHC.List.filter :: p:(a -> Bool) -> xs:[a] -> {v : [a] | v == GHC.List.filter p xs && len v <= len xs} @-}
+{-@ measure GHC.Real.even :: Integral a => a -> Bool @-}
+{-@ assume GHC.Real.even :: Integral a => x:a -> {VV : Bool | VV == GHC.Real.even x} @-}
+
+{-@ reflect keepEvens @-}
+keepEvens :: [Int] -> [Int]
+keepEvens = filter even
+```
+
+### Basic use
+
+There is no specific syntax for opaque reflection. Instead, it takes place when doing reflection, i.e. with `{-@ reflect keepEvens @-}`. During reflection, the plugin goes through all the variables used in the definition of the function to reflect and looks for unlifted (= not present in the logic environment) free variables. In the precedent example, these free variables are `filter` and `even`.
+
+For each unlifted free variable, it introduces a measure for it. For each measure introduced, we link the measure to the original function by strengthening the post-condition of the function, using the same pipeline as for reflection. We strengthen the post-condition by saying that the result `vv` is equal to the application of the measure to the arguments.
+
+If the user wishes to see what functions were opaque reflected, they can use the `--dump-opaque-reflections` pragma. It will list in the stdout the functions for which a measure was introduced automatically by opaque reflection.
+
+For instance, 
```

Comment:
This is an example of strengthening the post-condition. It should come before the paragraph about the flag `--dump-opaque-reflections`.